packages/wrangler/src/versions/index.ts

```diff
@@ -191,6 +215,21 @@ export async function versionsUploadHandler(
 		}
 	);
 
+	args.legacyAssets = args.legacyAssets ?? args.assets;
+
+	if (args.site || config.site) {
+		throw new UserError(
+			"Workers Sites are not supported in Gradual Deployments."
+		);
+	}
+	if (args.legacyAssets || config.legacy_assets) {
+		throw new UserError(
+			"Legacy Assets are not supported in Gradual Deployments."
+		);
+	}
```

Comment:
cc @irvinebroque @tanushree-sharma will this copy suffice?

Reply:
I would modify to "Workers Sites does not support uploading versions through "wrangler versions upload". You must use [wrangler deploy](https://developers.cloudflare.com/workers/wrangler/commands/#deploy) instead."

Reply:
Gah sorry, commenting after you've merged.

Reply:
Thanks for the copy. I'll make another PR to update